src/dstore/pmix_esh.c

```diff
@@ -574,62 +863,96 @@ static int _esh_patch_env(char ***env)
     return pmix_setenv(PMIX_DSTORE_ESH_BASE_PATH, _base_path, true, env);
 }
 
-static int _esh_nspace(const char *nspace)
+static int _esh_nspace(const char *nspace, pmix_info_t info[], size_t ninfo)
 {
     struct stat st = {0};
     seg_desc_t *seg;
+    int rc;
+    int tbl_idx;
+    nspace_tbl_entry_t *nspace_tbl_entry = NULL;
 
-    if (0 > asprintf(&_nspace_path, "%s/%s", _base_path, nspace)) {
-        PMIX_ERROR_LOG(PMIX_ERROR);
-        return PMIX_ERROR;
-    }
+    if (PMIX_SUCCESS != _esh_nspace_tbl_search(_jobuid, &tbl_idx)) {
 
-    if (stat(_nspace_path, &st) == -1){
-        if (0 != mkdir(_nspace_path, 0770)) {
+        rc = _esh_nspace_tbl_add(&tbl_idx);
+        if (PMIX_SUCCESS != rc) {
+            return rc;
+        }
+        nspace_tbl_entry = &_nspace_tbl[tbl_idx];
+
+        nspace_tbl_entry->jobuid = _jobuid;
```

Comment:
This is incorrect.
You have to check `info` keys first to see if another UID needs to be used. `_jobuid` is the default.